src/meta/src/manager/diagnose.rs

```diff
@@ -175,14 +176,35 @@ impl DiagnoseCommand {
                 worker_node.get_state().ok().map(|s| s.as_str_name()),
             );
             try_add_cell(&mut row, worker_node.parallelism());
-            try_add_cell(
-                &mut row,
-                worker_node.property.as_ref().map(|p| p.is_streaming),
-            );
-            try_add_cell(
-                &mut row,
-                worker_node.property.as_ref().map(|p| p.is_serving),
-            );
+            // is_streaming and is_serving are only meaningful for ComputeNode
+            let (is_streaming, is_serving) = {
+                if let Ok(t) = worker_node.get_type()
+                    && t == WorkerType::ComputeNode
+                {
+                    (
+                        worker_node.property.as_ref().map(|p| p.is_streaming),
+                        worker_node.property.as_ref().map(|p| p.is_serving),
+                    )
+                } else {
+                    (None, None)
+                }
+            };
+            try_add_cell(&mut row, is_streaming);
+            try_add_cell(&mut row, is_serving);
+            // is_iceberg_compactor is only meaningful for Compactor worker type
+            let is_iceberg_compactor = {
```

Comment:
should we change `rw_worker_nodes` table and show the `is_iceberg_compactor` flag?

https://github.com/risingwavelabs/risingwave/blob/6ebd5e5671551c9f3cbf2f20b3838cd5a976fda5/src/frontend/src/catalog/system_catalog/rw_catalog/rw_worker_nodes.rs#L44